apps/emqx_bridge_rabbitmq/src/emqx_bridge_rabbitmq_connector.erl

```diff
@@ -0,0 +1,557 @@
+                    default => <<"">>,
+                    desc => ?DESC("payload_template")
+                }
+            )}
+    ].
+
+values(post) ->
+    maps:merge(values(put), #{name => <<"connector">>});
+values(get) ->
+    values(post);
+values(put) ->
+    #{
+        server => <<"localhost">>,
+        port => 5672,
+        enable => true,
+        pool_size => 8,
+        type => rabbitmq,
+        username => <<"guest">>,
+        password => <<"guest">>,
+        routing_key => <<"my_routing_key">>,
+        payload_template => <<"">>
+    };
+values(_) ->
+    #{}.
+
+%% ===================================================================
+%% Callbacks defined in emqx_resource
+%% ===================================================================
+
+%% emqx_resource callback
+
+callback_mode() -> always_sync.
+
+-spec is_buffer_supported() -> boolean().
+
+%% emqx_resource callback
+
+is_buffer_supported() ->
+    %% We want to make use of EMQX's buffer mechanism
+    false.
+
+%% emqx_resource callback called when the resource is started
+
+-spec on_start(resource_id(), term()) -> {ok, resource_state()} | {error, _}.
+%% emqx_plugin_libs_pool
+on_start(
+    InstanceID,
+    #{
+        pool_size := PoolSize,
+        payload_template := PayloadTemplate
+    } = Config
+) ->
+    ?SLOG(info, #{
+        msg => "starting_rabbitmq_connector",
+        connector => InstanceID,
+        config => emqx_utils:redact(Config)
+    }),
+    Options = [
+        {config, Config},
+        %% The pool_size is read by ecpool and decides the number of workers in
+        %% the pool
+        {pool_size, PoolSize},
+        {pool, InstanceID}
+    ],
+    ProcessedTemplate = emqx_plugin_libs_rule:preproc_tmpl(PayloadTemplate),
+    State = #{
+        poolname => InstanceID,
+        processed_payload_template => ProcessedTemplate,
+        config => Config
+    },
+    case emqx_resource_pool:start(InstanceID, ?MODULE, Options) of
+        ok ->
+            {ok, State};
+        {error, Reason} ->
+            LogMessage =
+                #{
+                    msg => "rabbitmq_connector_start_failed",
+                    error_reason => Reason,
+                    config => emqx_utils:redact(Config)
+                },
+            ?SLOG(info, LogMessage),
+            {error, Reason}
+    end.
+
+%% emqx_resource callback called when the resource is stopped
+
+-spec on_stop(resource_id(), resource_state()) -> term().
+
+on_stop(
+    ResourceID,
+    #{poolname := PoolName} = _State
+) ->
+    ?SLOG(info, #{
+        msg => "stopping RabbitMQ connector",
+        connector => ResourceID
+    }),
+    Workers = [Worker || {_WorkerName, Worker} <- ecpool:workers(PoolName)],
+    Clients = [
+        begin
+            {ok, Client} = ecpool_worker:client(Worker),
+            Client
+        end
+     || Worker <- Workers
+    ],
+    %% We need to stop the pool before stopping the workers as the pool monitors the workers
```

Comment:
For my own information:

On the one hand, it seems we _could_ define an [`on_disconnect` callback](https://github.com/emqx/ecpool/blob/d01b8cb99af90bc177eeeabe29075133db878fb3/src/ecpool_worker.erl#L206-L207) with `ecpool`, perhaps that could be used?

On the other hand, it seems like the supervisees returned by `connect` [should be stopped](https://github.com/emqx/ecpool/blob/d01b8cb99af90bc177eeeabe29075133db878fb3/src/ecpool_worker.erl#L215) by ecpool?

Reply:
I guess it depends on the pool how the supervisees should be stopped but maybe that is what on_disconnect is for? Should we leave this as it is for now and fix it in a nicer way later if ecpool is extended (if the on_disconnect callaback has to be added)?

Reply:
It was more of a curiosity, if you happened to test either option.   I myself just discovered the `supervisees` API from your PR.  :smile_cat: 
If you prefer to change it later, sure.